File: androidApp/src/main/java/com/example/bmicalculator/android/common/ui/button/AppFlatButton.kt
package com.example.bmicalculator.android.common.ui.button

import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.material3.ButtonDefaults
import androidx.compose.material3.Text
import androidx.compose.material3.TextButton
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import com.example.bmicalculator.android.theme.LocalBmiColor
import com.example.bmicalculator.common.ui.model.AppFlatButtonUiState

@Composable
fun AppFlatButton(uiState: AppFlatButtonUiState) {
    TextButton(
        modifier = Modifier.fillMaxWidth(),
        enabled = uiState.enabled,
        onClick = uiState.onClick,
        colors =ButtonDefaults.textButtonColors(contentColor = LocalBmiColor.current.buttonEnableBackground)
    ) {
        Text(uiState.text)
    }
}

File: web_app/bmi_cal_web_app/src/common/AppFlatButton.js
import { Button } from "@mui/material"
import kmp from "BmiCalculator-shared"

kmp.com.example.bmicalculator.common.ui.model.AppFlatButtonUiState.prototype.toComponent = function () {
    return <div style={{ marginLeft: "16px", marginRight: "16px" }}>
        <Button
            variant="text"
            onClick={this.onClick}
            disabled={!this.enabled}
            fullWidth>
            {this.text}
        </Button>
    </div>
}
